rfc/cdevents-translator.md

```diff
@@ -0,0 +1,41 @@
+## CDEvents Translator
+
+### Overview:
+This proposal describes translating events from various source control systems into CDEvents and sending them to configured message-broker, by using a common cdevents-translator 
+
+A library cdevents-translator will be developed to receive various event types and convert them into CDEvents.
+
+Various types of source code version control systems like GitHub, Gerrit and Gitlab endpoints/webhooks can be configured and implemented with in the cdevents-translator.
+
+Instructions for configuring Gerrit webhooks and implementation details can be found in [gerrit-cdevent](gerrit-cdevents.md) 
+
+### Design approach
+Create a Golang application with an API server configured to initialize SCM systems webhooks/endpoints to receive events and translate them into CDEvents.
```

Comment:
The following just for info.

Tekton Triggers provides and API server that can reply to webhooks. It supports incoming `CloudEvents` as well as generic webhooks and it provides a powerful templating mechanism to extract information from incoming payloads and produce a YAML out of them. I used it in this POC https://github.com/afrittoli/cdeventer.

The approach of CDEventer is a bit more complex, but it has the advantage of decoupling the processing of incoming messages, which is then implemented as a set of input specific `TriggerBindings`, from the actual sending of the target CDEvents, through the use of an intermediate YAML.

The downside of CDEventer is the extra dependency to Tekton (and thus K8s).